saito-core/src/core/routing_event_processor.rs

```diff
@@ -153,50 +120,15 @@ impl RoutingEventProcessor {
         debug!("incoming message processed");
     }
 
-    // async fn propagate_block_to_peers(&self, block_hash: SaitoHash) {
-    //     debug!("propagating blocks to peers");
-    //     let buffer: Vec<u8>;
-    //     let mut exceptions = vec![];
-    //     {
-    //         trace!("waiting for the blockchain write lock");
-    //         let blockchain = self.blockchain.read().await;
-    //         trace!("acquired the blockchain write lock");
-    //         let block = blockchain.blocks.get(&block_hash);
-    //         if block.is_none() {
-    //             // TODO : handle
-    //         }
-    //         let block = block.unwrap();
-    //         buffer = block.serialize_for_net(BlockType::Header);
-    //
-    //         // finding block sender to avoid resending the block to that node
-    //         if block.source_connection_id.is_some() {
-    //             trace!("waiting for the peers read lock");
-    //             let peers = self.peers.read().await;
-    //             trace!("acquired the peers read lock");
-    //             let peer = peers
-    //                 .address_to_peers
-    //                 .get(&block.source_connection_id.unwrap());
-    //             if peer.is_some() {
-    //                 exceptions.push(*peer.unwrap());
-    //             }
-    //         }
-    //     }
-    //
-    //     self.io_handler
-    //         .send_message_to_all(buffer, exceptions)
-    //         .await
-    //         .unwrap();
-    //     debug!("block sent to peers");
-    // }
-
     async fn connect_to_static_peers(&mut self) {
         debug!("connect to peers from config",);
         trace!("waiting for the configs read lock");
         let configs = self.configs.read().await;
         trace!("acquired the configs read lock");
 
         for peer in &configs.peers {
-            self.io_interface
+            self.network
+                .io_interface
                 .connect_to_peer(peer.clone())
                 .await
                 .unwrap();
```

Comment:
Isn't it better to move this implementation to the network struct itself and call the _connect to static peers_ method of the network struct from here?

Reply:
yes will do.